src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1202,41 +1202,21 @@ public function resolveIdenticalType(Type $leftType, Type $rightType): BooleanTy
 		}
 
 		if ($leftType instanceof ConstantArrayType && $rightType instanceof ConstantArrayType) {
-			return $this->resolveConstantArrayTypeComparison($leftType, $rightType, fn ($leftValueType, $rightValueType): BooleanType => $this->resolveIdenticalType($leftValueType, $rightValueType));
+			return self::resolveConstantArrayTypeComparison($leftType, $rightType, fn ($leftValueType, $rightValueType): BooleanType => $this->resolveIdenticalType($leftValueType, $rightValueType));
 		}
 
 		return new BooleanType();
 	}
 
 	public function resolveEqualType(Type $leftType, Type $rightType): BooleanType
```

Comment:
if desired, `resolveEqualType` could be inlined as it is now a single line method